Automattic/woocommerce-payments
# Add light `npm run up` for quickly starting up docker environment with any config changes

Fixes #2915

#### Changes proposed in this Pull Request
This PR makes two changes to npm scripts for wrangling the docker dev environment:

- Rename `npm run up` => `up:recreate` so it's clear it will recreate everything and may take a while. 
- Reuse `npm run up` for lightweight re-up when needed (e.g. tweak mapped plugins, or after a reboot). 

This should not hurt any dev flows. Anyone using `npm run up` will get a quicker `up`, and if they want to recreate (as per current script) they can use `npm run up:recreate` when needed.

I'm open to feedback on any of this, especially the naming.

The main thing I want is a quick way to pick up compose config changes, specifically mapped folders. The new `npm run up` works great for this, and I believe it should work well for other dev use cases.

#### Testing instructions
* Set up docker dev env as per readme.
* While there, [review the changes to `docker/README.md`](https://github.com/Automattic/woocommerce-payments/pull/2916/files?short_path=da8fcbe#diff-da8fcbe728a9172b578e5d754f8e2df214c658c4321f610e63dd68bea828ab49) and check that they are clear and helpful.
* Switch to this branch (if not there already).
* Add a `docker-compose.override.yml` file or make changes to `docker-compose.yml` file, e.g. add a new mapped folder (see example below).
* `npm run up` 
* Confirm the docker env includes the changed config 🎉 
* Ensure `npm run up:recreate` still works correctly for initial setup and recreate / reset.

```
# docker-compose.override.yml
version: '3'

# Override services.wordpress.volumes to map additional plugin repos for cross-project debugging.
services:
  wordpress:
    volumes:
      - ../woocommerce-subscriptions:/var/www/html/wp-content/plugins/woocommerce-subscriptions
      - ../woocommerce-blocks:/var/www/html/wp-content/plugins/woocommerce-blocks
```

-------------------

- [x] Added changelog entry (or does not apply) n/a - dev only
- [x] Covered with tests (or have a good reason not to test in description ☝️) n/a - dev only
- [x] Tested on mobile (or does not apply) n/a - dev only

**Post merge**



- [x] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply) n/a - dev only
